apps/marketplace/components/Opportunities/Opportunities.js

```diff
@@ -0,0 +1,310 @@
+                      <div className={`${styles.badge} ${item.openTo === 'all' ? styles.green : styles.blue}`}>
+                        {item.openTo}
+                      </div>
+                    </div>
+                    <div className={`col-md-1 col-sm-1 ${styles.cell} ${styles.alignCenter}`}>{item.id}</div>
+                    <div className={`col-md-4 col-sm-4 ${styles.cell}`}>
+                      <a href={`/digital-marketplace/opportunities/${item.id}`}>{item.name}</a>
+                      <div>At: {item.company}</div>
+                    </div>
+                    <div className={`col-md-2 col-sm-2 ${styles.cell}`}>{item.location}</div>
+                    <div className={`col-md-2 col-sm-2 ${styles.cell}`}>{item.closing}</div>
+                    <div className={`col-md-1 col-sm-1 ${styles.cell} ${styles.lastColumn}`}>{item.submissions}</div>
+                  </div>
+                </div>
+              ))}
+            </div>
+          )}
+          {items.length > 0 && (
+            <div className={`${styles.tableMobile} ${styles.hideDesktop}`}>
+              {items.map(item => (
+                <div className={styles.tableRowMobile} key={`item.${item.id}`}>
+                  <div className="row">
+                    <div className={`col-md-10 col-sm-10 col-xs-11 ${styles.mobileColumn}`}>
+                      <div className={`${styles.badgeMobile} ${item.openTo === 'all' ? styles.green : styles.blue}`}>
+                        open to {item.openTo.toLowerCase()}
+                      </div>
+                      <div className={styles.mobileName}>
+                        <a href={`/digital-marketplace/opportunities/${item.id}`}>{item.name}</a>
+                        <div className={styles.companyMobile}>At: {item.company}</div>
+                      </div>
+                      <div className={`${styles.mobileDetails} row`}>
+                        <div className="col-md-3 col-sm-3 col-xs-4">
+                          <div>
+                            <strong>Location:</strong>
+                          </div>
+                          <div>
+                            <strong>Closing:</strong>
+                          </div>
+                          <div>
+                            <strong>Submissions:</strong>
+                          </div>
+                        </div>
+                        <div className="col-md-9 col-sm-9 col-xs-8">
+                          <div>{item.location}</div>
+                          <div>{item.closing}</div>
+                          <div>{item.submissions}</div>
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              ))}
+            </div>
+          )}
+        </article>
+      </div>
+    )
+  }
+}
+
+Opportunities.propTypes = {}
+
+const mapStateToProps = ({}) => ({})
+
+export default connect(mapStateToProps)(Opportunities)
```

Comment:
Does this need to be a connected component? the `mapStateToProps` is empty so it doesn't seem like it's actually using any redux state?

Reply:
Yes! This is just the UI part, the redux part is missing and that is where this will come in handy. Redux part will be fun :)

Reply:
Ah of course! sorry I forgot it still needs backend integration.

For the person doing the integration: I still suspect the opportunities component won't need to be connected to redux, as the page will do the network call(s) and can pass in the items as a prop.

Reply:
Stateless components are preferred when there is no state management in the component. However this component will require state management because it has so many state changes in the accordion and filtering. Ultimately stateless vs stateful is just a presentation choice, don't affect anything else. Same as composition of dividing a component into subcomponents. Just helps readability. But give time constrains it is all one big blob :P